src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,127 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.command.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Represents a parser that processes input command lines from a user.
+ */
+public class Parser {
+    /**
+     * Parses the input line from a user.
+     * @param commandLine Input string to be parsed.
+     * @return A command based on the input.
+     * @throws DukeException If invalid command is given.
+     */
+    public static Command parse(String commandLine) throws DukeException{
+        String[] messages = commandLine.split(" ");
+        String commandType = messages[0];
+
+        switch (commandType) {
+        case "bye":
+            return new ExitCommand();
+
+        case "delete":
+            if (messages.length == 1) {
+                throw new DukeException("The task index is missing.");
+            }
+            return new DeleteCommand(Integer.parseInt(messages[1]));
+
+        case "done":
+            if (messages.length == 1) {
+                throw new DukeException("The task index is missing.");
+            }
+            int id = Integer.parseInt(messages[1]);
+            return new DoneCommand(id);
+
+        case "deadline":
+            if (messages.length == 1) {
+                throw new DukeException("The task description is missing.");
+            }
+            try {
+                return generateDeadline(processTask(commandLine));
+            } catch (Exception ex) {
+                throw new DukeException("Deadline description is in wrong format.");
+            }
+
+        case "event":
+            if (messages.length == 1) {
+                throw new DukeException("The task description is missing.");
+            }
+            try {
+                return generateEvent(processTask(commandLine));
+            } catch (Exception ex) {
+                throw new DukeException("Event description is in wrong format.");
+            }
+
+        case "todo":
+            if (messages.length == 1) {
+                throw new DukeException("The task description is missing.");
+            }
+            return new AddTodo(new Todo(processTask(commandLine), 0));
+
+        case "list":
+            return new ListCommand();
+
+        case "find":
+            if (messages.length == 1) {
+                throw new DukeException("The keyword is missing.");
+            }
+            return new FindCommand(processTask(commandLine));
+
+        default:
+            throw new DukeException("I'm sorry, but I don't know what " + commandType + " means.");
+
+        }
+    }
+
+    /**
+     * Processes the String to get the task description.
+     * @param message Input string to be processed.
+     * @return A task description.
+     */
+    public static String processTask(String message){
+        return message.substring(message.indexOf(' ') + 1);
```

Comment:
Does the 1 here look a bit like a magic number? Perhaps can consider explaining it in the comment